training/prepare_dataset.py

```diff
@@ -287,11 +287,13 @@ def compute_prompt_embeddings(
 
 
 def save_image(image: torch.Tensor, path: pathlib.Path) -> None:
-    image = to_pil_image(image)
+    image = image.to(dtype=torch.float32).clamp(-1, 1)
+    image = to_pil_image(image.float())
     image.save(path)
```

Comment:
Could you explain this change?

Reply:
F.ToPILImage does not support bfloat16. So if someone was to do precomputation in bfloat16 and try to save the images/videos in viewable format, it would error out. casting to fp32 fixes this